app/src/main/java/com/andrew/tasky/auth/util/GetResourceResult.kt

```diff
@@ -0,0 +1,47 @@
+package com.andrew.tasky.auth.util
+
+import com.andrew.tasky.R
+import com.andrew.tasky.core.ErrorMessageDto
+import com.andrew.tasky.core.Resource
+import com.squareup.moshi.Moshi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.ResponseBody
+import okio.IOException
+import retrofit2.HttpException
+import retrofit2.Response
+
+suspend fun <T> getResourceResult(apiToBeCalled: suspend () -> Response<T>): Resource<T> {
+    return withContext(Dispatchers.IO) {
+        try {
+            val response = apiToBeCalled()
```

Comment:
If you return a `Response` here, then afaik this doesn't throw an exception and instead includes the error in the response result, so the try catch isn't necessary if I remember correctly